docs/proposal-0007-discrete_metric_execution.md

```diff
@@ -0,0 +1,107 @@
+# Reducing Jetstream Resource Usage Errors with Discrete Metric Execution
+
+* Status: accepting feedback
+* Deciders: mwilliams, ascholtz, dberry
+
+
+## Context and Problem Statement
+
+Periodically, Jetstream fails during automated analysis of experiments. Often, this is due to resource limits, which occur in either BigQuery or the Jetstream cluster. Each failure requires manual debugging from DE and/or DS to determine the issue and come up with a solution. Often, the required solution is suboptimal and requires substantial compromise. For example, common solutions include removing metrics from the experiment, which leads to less information from the experiment, and downsampling the experiment population, which leads to lower confidence/precision to results.
+
+### Other Considerations
+There are many improvements we would like to make to analysis that could be enabled depending on the approach we take to these Jetstream architecture changes, for example:
+- **Covariate adjustment**
+  - requires some analysis periods to be dependent on others
+  - caveat: preenrollment analysis periods may never be available (e.g., new user experiments)
```

Comment:
I think in this case we'd still want later analysis periods to block on the preenrollment period step being completed, it's just that that step may complete quickly or with empty output. The following periods should be robust to invalid preenrollment data, but they should be able to assume that the preenrollment step has completed.